### PR TITLE
Add ability to define queue timeout in scheduler workload

### PR DESCRIPTION
**Pull request by hasit**
### Description of change

screencapture-localhost-3000-d-1cafa24eadc87e0eeb9217b9179e31473bcfa21d-aperture-service-protection-2023-07-06-14_37_41

##### Checklist

- [x] Tested in playground or other setup
- [x] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added
- [ ] Tests and/or benchmarks are included
- [ ] Breaking changes

### Summary by CodeRabbit

```markdown
**New Feature**
- Added a new field `queue_timeout` to the `Scheduler` message in protobuf definition, allowing users to set a timeout for the flow in the workload. The minimum of the client's timeout and the defined workload timeout is used.

**Refactor**
- Simplified function signatures of `ProcessRequest` in the `Engine` struct and `CheckRequest` in the `Handler` struct by writing parameters on one line.
```

> 🎉 With timeouts now in control, our flows are on a roll! 🚀
> No more waiting in despair, your requests are handled with care. 💻
> Function signatures sleek and neat, making our codebase petite! 📚

### CODE REVIEW

**Review by tanveergill — APPROVED** [review]

File: pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go [hunk 1]
```diff
@@ -411,7 +420,8 @@ func (s *Scheduler) Decide(ctx context.Context, labels labels.Labels) iface.Limi
 
 	reqCtx := ctx
 
-	if clientDeadline, hasDeadline := ctx.Deadline(); hasDeadline {
+	clientDeadline, hasDeadline := ctx.Deadline()
+	if hasDeadline || hasWorkloadTimeout {
```
Comment (tanveergill) [inline pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go:424]: `clientDeadline` is valid only when `hasDeadline == true`. Cannot make the calculation at https://github.com/fluxninja/aperture/pull/2326/files#diff-9bad7db7432925c3ff46ac145c61f11d0733bc81463f2c6c85c082f95b4f8e95R431

File: pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go [hunk 2]
```diff
@@ -425,6 +435,12 @@ func (s *Scheduler) Decide(ctx context.Context, labels labels.Labels) iface.Limi
 			// dropped if it doesn't get the tokens immediately.
 			timeout = 0
 		}
+
+		// find the minimum of matchedWorkloadTimeout and client's timeout
+		if hasWorkloadTimeout && matchedWorkloadTimeout < timeout {
```
Comment (tanveergill) [inline pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go:440]: @hasit,

This doesn't cover the scenario when `clientDeadline` is not defined (0) and workload timeout is defined

Comment (tanveergill) [reply pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go:440]: Also, the behavior might be unexpected in the reverse scenario when `clientDeadline` is defined and workload timeout is not defined

File: pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go (hunk 1 above)
Comment (hasit) [reply pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go:424]: Addressed in https://github.com/fluxninja/aperture/pull/2336.

File: pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go (hunk 2 above)
Comment (hasit) [reply pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go:440]: Addressed in https://github.com/fluxninja/aperture/pull/2336.

**Pull request merged**